### ISSUE TITLE
[🐞]  function of `ref` attribute of a dynamic tag does not called

### ISSUE THREAD

**Issue by mechairoi** [BuilderIO/qwik#5660]
### Which component is affected?

Qwik Runtime

### Describe the bug

When setting a function to the `ref` attribute of a dynamic tag, the function does not get called. This was working up until version 1.2.13, but it seems to have stopped working from version 1.2.14.

```tsx
export const Foo = component$((props: { isDiv?: boolean }) => {
  const element = useSignal<Element>();
  const TagName = props.isDiv ? 'div' : 'p';
  return (
    <TagName
      ref={(el) => {
        console.log(el);
        element.value = el;
      }}
    >
      element nodeType is {element.value?.nodeType}
    </TagName>
  );
});
```

### Reproduction

https://stackblitz.com/edit/qwik-starter-7urfnz?file=src%2Froutes%2Findex.tsx,package.json

### Steps to reproduce

`npm install && npm run dev`

### System Info

```shell
System:
    OS: macOS 13.6.3
    CPU: (8) arm64 Apple M1
    Memory: 156.70 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.5.0 - ~/.local/share/rtx/installs/node/21.5.0/bin/node
    Yarn: 1.22.19 - ~/.local/share/rtx/installs/node/21.5.0/bin/yarn
    npm: 10.2.4 - ~/.local/share/rtx/installs/node/21.5.0/bin/npm
    pnpm: 8.13.1 - ~/.local/share/rtx/installs/node/21.5.0/bin/pnpm
    bun: 1.0.20 - ~/.local/share/rtx/installs/bun/1.0.20/bin/bun
  Browsers:
    Brave Browser: 107.1.45.127
    Chrome: 120.0.6099.129
    Safari: 17.2
    Safari Technology Preview: 17.4
  npmPackages:
    @builder.io/qwik: ^1.3.1 => 1.3.1 
    @builder.io/qwik-city: ^1.3.1 => 1.3.1 
    @builder.io/qwik-react: 0.5.0 => 0.5.0 
    undici: ^5.26.0 => 5.28.2 
    vite: ^5.0.6 => 5.0.10
```

### Additional Information

This issue seems to prevent client-only components in qwik-react from `render`ing.

https://github.com/BuilderIO/qwik/blob/72009bbf1af6ce506c1367dcde098fa76c6fc48c/packages/qwik-react/src/react/qwikify.tsx#L96-L98

**Comment by wmertens** [comment]
This will probably be fixed by #5650

**Comment by mechairoi** [comment]
I have confirmed it is fixed in v1.3.3. Thank you!
